### PR TITLE
python: migrate data source to OSV

### PR DESCRIPTION
**Pull request by daynewlee**

### CODE REVIEW

**Comment by RTann** [comment]
I'm cool with it, but up to @crozzy and @hdonnay if pyup.io support should be removed. If we do end up removing it, then let's add the `needs-changelog` label to this PR

File: python/matcher.go
Comment (RTann) [inline python/matcher.go:22]: I think the `pep440` check still makes sense for this, but I'm not 100% sure. I'll defer this to someone else

File: python/matcher.go
Comment (RTann) [inline python/matcher.go:44]: I'm not sure if pypi makes sense here. wouldn't this be an osv problem?

File: test/periodic/updater_test.go
Comment (RTann) [inline test/periodic/updater_test.go:1]: Should we replace the removed one with and OSV test?

File: python/testdata/indexreport-buster-jackson-databind.json
Comment (RTann) [inline python/testdata/indexreport-buster-jackson-databind.json:1]: can we rename this to rhel8 or something? this is definitely not buster (debian 10)

File: python/matcher_test.go
Comment (RTann) [inline python/matcher_test.go:5]: nit: put this in the block directly below with zlog

File: python/matcher_test.go
Comment (RTann) [inline python/matcher_test.go:43]: can you add a test case for a package version less than the "introduced" version?

File: python/matcher.go
Comment (RTann) [inline python/matcher.go:52]: nit: maybe `uv`?

File: python/matcher.go
Comment (RTann) [inline python/matcher.go:6]: let's keep the logs. It can be helpful to log any parsing failures

File: updater/osv/osv.go
Comment (RTann) [inline updater/osv/osv.go:601]: nit: maybe a switch here, too? might be more readable and extensible than the if in this context

File: python/matcher.go
Comment (hdonnay) [reply python/matcher.go:22]: Yeah, I think it still does, even if only in a belt-and-suspenders way.

File: python/testdata/indexreport-buster-rhel8-databind.json
Comment (hdonnay) [inline python/testdata/indexreport-buster-rhel8-databind.json:1]: What's up with this name? Why "buster" and "rhel8" -- is it both Debian and RHEL? What does "databind" mean, here?

**Review by hdonnay — CHANGES_REQUESTED** [review]
The first commit needs to be split into adding functionality to `osv` and removing `pyupio`.

We'll need a follow-up to clean up the leaked vulnerabilities, after this.

File: python/matcher_test.go
Comment (hdonnay) [inline python/matcher_test.go:9]: Fix grouping -- once rebased on latest `main`, `git rebase -x etc/importfix.sh origin/main` should do it

**Review by hdonnay — CHANGES_REQUESTED** [review]
LGTM after the one formatting nit.

**Comment by daynewlee** [comment]
Applied changes in etc/importfix.sh, then ran:
git rebase -x './etc/importfix.sh.patched; git diff --quiet || git commit -a -n --no-edit --fixup=HEAD' upstream/main

File: python/matcher.go
Comment (RTann) [inline python/matcher.go:24]: Let's just completely revert this part back since no update is needed

File: python/matcher.go
Comment (RTann) [inline python/matcher.go:45]: thinking about it, maybe this should say `python` instead of `osv` because, even though the data is OSV, the function is under the `python` directory. I'll let someone else chime in about this because I'm not 100% sure the convention used in this repo

Comment (crozzy) [reply python/matcher.go:45]: agree

**Review by hdonnay — APPROVED** [review]
LGTM

**Pull request merged**
